src/app/loadout-builder/process.ts

```diff
@@ -416,19 +416,40 @@ export function generateMixesFromPerks(
   return mixes;
 }
 
-function getBaseStatValue(stat: DimStat, item: DimItem) {
-  let baseStatValue = stat.value;
+function getPlugHashesFromCategory(category: DimSocketCategory) {
+  return category.sockets
+    .map((socket) => {
+      return idx(socket, (socket) => socket.plug.plugItem.hash) || null;
+    })
+    .filter(Boolean);
+}
+
+function getBaseStatValues(stats: Dictionary<DimStat>, item: DimItem) {
+  const baseStats = {};
+  _.forEach(statValues, (statHash) => (baseStats[statHash] = stats[statHash].value));
```

Comment:
We shouldn't use `_.forEach` - you either want a normal for loop, or `_.forIn`, or some combination of `Object.keys`/`Object.values`.

Reply:
We use these a lot at work for IE support, I'll endeavour to not use them.

Reply:
Babel makes quick work of them even if you need to support an older version (via babel-preset-env)